### PR TITLE
Add asan/tsan docker build script

### LINKED ISSUE #8190: Add docker build script for asan/tsan testing

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #8190 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Manually test tsan build script.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

File: release-centos7-llvm/scripts/build-tiflash-release.sh [hunk 1]
```diff
@@ -74,13 +80,19 @@ cmake -S "${SRCPATH}" \
...
-
-# unset LD_LIBRARY_PATH before test
-unset LD_LIBRARY_PATH
-readelf -d "${INSTALL_DIR}/tiflash"
-ldd "${INSTALL_DIR}/tiflash"
-
-# show version
-${INSTALL_DIR}/tiflash version
+if [ $CMAKE_BUILD_TYPE == "TSAN" ] || [ $CMAKE_BUILD_TYPE == "ASAN" ]; then
+  cmake --build . --target gtests_dbms --parallel ${NPROC}
```
Comment (JaySon-Huang) [inline release-centos7-llvm/scripts/build-tiflash-release.sh:84]:
```suggestion
  cmake --build . --target gtests_dbms gtests_libcommon --parallel ${NPROC}
```
Could also build `gtests_libcommon`

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Pull request merged**